### DISCUSSION TITLE
Rebinding `cmd-j` and `cmd-k` to navigate in menus?

### DISCUSSION THREAD

**Discussion by nelsonr** [zed-industries/zed#11705] [Feature Requests]
Hi, 

I'm trying to rebind the keys to navigate up and down on the menus (e.g. "Search project files..."). 

I managed to rebind `cmd-j` to move down but trying to rebind `cmd-k` to move up never seems to work. This is what I have so far. 

```
{
  "bindings": {
    "cmd-j": "menu::SelectNext",
    "cmd-k": "menu::SelectPrev"
  }
}
```

I tried to check if there was a specific context for the Menu but it doesn't seem that there's one.

**Comment by Moshyfawn** [comment]
I believe this is because `cmd-k` is the first "chord" keypress: I modifier, if you will, to allow another layer of keypresses like `cmd-k cmd-w` _(2 chords)_.

**Reply by nelsonr** [reply]
Yes, that was also what I thought. 

Meanwhile, I switched it to use `ctrl-j` and `ctrl-k`, which is fine as well.

It would be nice to have the option of overriding it in specific contexts.

**Reply by nelsonr** [reply]
After checking it again. I noticed that it actually works, but since there's a delay, due to it being a chord, it's not immediately noticeable. It also works for context menus, with the same behavior.

**Comment by nelsonr** [comment]
Ok, so I believe I found a solution.

In order to be possible to use `cmd-k` as its own keybinding, it requires disabling all the default chords where it's used. This can be done by copying all instances of it from the default keymap into your keymap and then setting everything to `null`.

With this, Zed will no longer introduce a delay when using `cmd-k`.

```
[
  {
    "context": "Workspace",
    "bindings": {
      "cmd-k s": null,
      "cmd-k cmd-s": null,
      "cmd-k cmd-t": null,
      "cmd-k m": null,
      "cmd-k cmd-left": null,
      "cmd-k cmd-right": null,
      "cmd-k cmd-up": null,
      "cmd-k cmd-down": null,
      "cmd-k shift-left": null,
      "cmd-k shift-right": null,
      "cmd-k shift-up": null,
      "cmd-k shift-down": null
    }
  },
  {
    "context": "Pane",
    "bindings": {
      "cmd-k u": null,
      "cmd-k cmd-w": null,
      "cmd-k up": null,
      "cmd-k down": null,
      "cmd-k left": null,
      "cmd-k right": null
    }
  },
  {
    "context": "Editor",
    "bindings": {
      "cmd-k cmd-d": null,
      "cmd-k ctrl-cmd-d": null,
      "cmd-k cmd-i": null
    }
  },
  {
    "context": "Editor && mode == full",
    "bindings": {
      "cmd-k enter": null
    }
  },
]
```

**Reply by sipec** [reply]
this doesn't appear to work anymore, even when replacing all the commands in the most recent default keybindings :(

**Comment by favetelinguis** [comment]
For info anyone finding here this works for me when having this issue in the terminal for some TUI apps (broot):

```
  {
    "context": "Terminal",
    "bindings": {
      "ctrl-h": ["terminal::SendKeystroke", "ctrl-h"],
      "ctrl-j": ["terminal::SendKeystroke", "ctrl-j"],
      "ctrl-k": ["terminal::SendKeystroke", "ctrl-k"],
      "ctrl-l": ["terminal::SendKeystroke", "ctrl-l"],
    }
```
